packages/rocketchat-lib/server/functions/saveUser.js

```diff
@@ -173,6 +173,41 @@ RocketChat.saveUser = function(userId, userData) {
 
 		return _id;
 	} else {
+		if (!RocketChat.settings.get('Accounts_AllowUserProfileChange')) {
+			throw new Meteor.Error('error-action-not-allowed', 'Edit user profile is not allowed', {
+				method: 'insertOrUpdateUser',
+				action: 'Update_user'
+			});
+		}
+
+		if (userData.username && !RocketChat.settings.get('Accounts_AllowUsernameChange')) {
+			throw new Meteor.Error('error-action-not-allowed', 'Edit username is not allowed', {
+				method: 'insertOrUpdateUser',
+				action: 'Update_user'
+			});
+		}
+
+		if (userData.name && !RocketChat.settings.get('Accounts_AllowRealNameChange')) {
+			throw new Meteor.Error('error-action-not-allowed', 'Edit user real name is not allowed', {
+				method: 'insertOrUpdateUser',
+				action: 'Update_user'
+			});
+		}
+
+		if (userData.name && !RocketChat.settings.get('Accounts_AllowEmailChange')) {
```

Comment:
Looks like you made a little copy&paste error here. ;) Should be `userData.email` I guess.